src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/FileImportControl.java

```diff
@@ -55,16 +55,6 @@ class FileImportControl extends AbstractImportControl {
         }
     }
 
-    /**
-     * Enclose {@code expression} in a (non-capturing) group.
-     *
-     * @param expression the input regular expression
-     * @return a grouped pattern.
-     */
-    private static String encloseInGroup(String expression) {
-        return "(?:" + expression + ")";
```

Comment:
@rnveach , This code.


If we don't remember why we did this, we will remove it

Reply:
The comment above its call says `ensure that fullPackage is a self-contained regular expression`.

You had to search a bit more in the commit you found.
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java#L155-L156 was renamed from `ImportControl`, and then from `PkgControl` before that, and that is where the code originally came from. I did not add it.

It appears the code really came from https://github.com/checkstyle/checkstyle/commit/63e8c1dd4e1da443c8533d40640fe1a3b371afe8#diff-f68bc4c425df6a25439881dbcd97576b489f7d71fc7706a450e11a101f0d3e77R148-R149 .
https://github.com/checkstyle/checkstyle/issues/2999

Some things of interest:
https://github.com/checkstyle/checkstyle/pull/3438#issuecomment-252503843
https://github.com/checkstyle/checkstyle/pull/3438#discussion_r82990960
https://github.com/checkstyle/checkstyle/pull/3438#discussion_r83542445

Reply:
Ok, thanks a lot, after reading all of this, there is no special reason of such code.

Last reason could be performance, as regexp does not need do creation of group. But as we wrap whole regexp it is does not guarantee that all nested groups will be downgraded to non-captured groups.

I am ok to remove this method.

Reply:
@romani 

> (?:.*(Panel|View|Dialog))

What happens when the regexp changes to `.*(Panel|View|Dialog)`? Since this now a capturing group, will it cause any issues?

Reply:
If no code deals with group, no changes.
We do not use group, we just match, true or false.

Reply:
@romani To me the following lines explain why we need `encloseInGroup`.
https://github.com/checkstyle/checkstyle/blob/4d113f93dd9a4a731515f2e5da76112e37afd49b/src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/PkgImportControl.java#L128-L130

We can't rely on the users adding `()` themselves and we enclose it in a `()` in case they don't.

Reply:
> If no code deals with group, no changes.
We do not use group

Or if we do not combine 2 regexps.

Reply:
@romani https://github.com/checkstyle/checkstyle/pull/13760#discussion_r1337408244 . I also approved this PR as I think the code removal is fine.